assets/js/modules/reader-revenue-manager/components/common/PublicationCreate.js

```diff
@@ -43,21 +39,27 @@ export default function PublicationCreate( { onCompleteSetup } ) {
 	const publications = useSelect( ( select ) =>
 		select( MODULES_READER_REVENUE_MANAGER ).getPublications()
 	);
-
 	const serviceURL = useSelect( ( select ) =>
 		select( MODULES_READER_REVENUE_MANAGER ).getServiceURL()
 	);
 
-	if ( publications === undefined ) {
-		return null;
-	}
+	const { selectPublication } = useDispatch( MODULES_READER_REVENUE_MANAGER );
 
 	const hasPublication = publications && publications.length > 0;
 
-	const handleCompleteSetupClick = () => {
-		// TODO: Track event for completing setup.
+	const handleCompleteSetupClick = useCallback( async () => {
+		if ( ! hasPublication ) {
+			return;
+		}
+
+		await selectPublication( publications[ 0 ] );
```

Comment:
The change in this component ensures that the created publication is selected in store before saving settings.